src/datastores/sql/models/file.py

```diff
@@ -172,6 +173,17 @@ def path(self):
             filename = f"{filename}.{self.extension}"
         return os.path.join(self.folder.path, filename)
 
+    def purge(self, db):
+        # Check if the file is soft deleted before purging
+        if not self.is_deleted:
+            raise Exception("File must be deleted before purging")
+
+        if os.path.exists(self.path):
```

Comment:
The pre-condition here is that ```self.path``` is an actual file, and it will never be a directory, right?

Reply:
Yeah, that's right. This is on the File model object so it will always be a path to a file and not a directory.